cherrypy/lib/static.py

```diff
@@ -75,6 +75,13 @@ def serve_file(path, content_type=None, disposition=None, name=None,
         i = path.rfind('.')
         if i != -1:
             ext = path[i:].lower()
+            if ext in mimetypes.encodings_map.keys():
```

Comment:
Could you please describe the logic you are adding in more detail? It should be possible to read and understand this code easily in the future.